### ISSUE TITLE
Unable to use variables

### ISSUE THREAD

**Issue by jplambert** [smartrics/RestFixture#110]
Hi,

I have spent most of the day trying very hard to use variables with RestFixture and other fixtures and have been unable so far to make it work completely. Could you please help me?

This is what I do: (code simplified for the purpose of showing)

|Table: Rest Fixture|http://localhost||
|GET|/ws/getId.php?filter=interesting|200|Content-Type : application/json||
|let|id|js|${jsonbody}.items[0].id||

Using a custom fixture to compute SHA-256 hash, used as a check for the next webservice.

|Table: Hash Fixture|SHA-256|
|sometext$id|$computedkey=|

|Table: Rest Fixture|http://localhost||
|GET|/ws/getDetails.php?key=$computedkey&id=$id|200|Content-Type : application/xml; charset=UTF-8|'![CDATA[OK]]'|

What do I see?

1. My custom "Hash Fixture" gets the "sometext$id" String passed to it, instead of what has been found out by the first "Rest Fixture" table concatenated to "sometext". Please note that I also tried with only "$id" but got the same result: the "$id" String gets passet to the custom "Hash Fixture" instead of what has been found out by the first "Rest Fixture" table.
2. The second "Rest Fixture" table resolves succesfully "$computedKey" but keeps "$id" as if.

To me, it looks like as if the first "Rest Fixture" table do not set the $id variable.

However, I have another test instance where it actually works: a first "Rest Fixture" table sets a variable" which is used in another "Rest Fixture" table. So I find this issue all the more disturbing.

One important note: I am using Slim. And would like not to change since it makes easier to implement my custom "Hash Fixture" and because I am also testing PHP for which FitNesse runner is only available for Slim.

However, I also quickly tried using Fit, and it didn't work either...

Am I doing something wrong, or is there some bug or unfinished code lurking somewhere?

Thanks in advance,

Jean-Pierre Lambert

**Comment by smartrics** [comment]
Hi, 
should this 
|Table: Rest Fixture|http://localhost||
|GET|/ws/getId.php?filter=interesting|200|Content-Type : application/json||
|let|id|js|${jsonbody}.items[0].id||

be actually 

|Table: Rest Fixture|http://localhost||
|GET|/ws/getId.php?filter=interesting|200|Content-Type : application/json||
|let|id|js|response.jsonbody.items[0].id||

Please refer to issue #52. I don't have bandwidth to fix it but i will be happy to merge and release the rest fixture if you do contribute a fix via pull request.

your other option is to extend the RestFixture, add your method to calculate the hash and use your extension instead.

**Comment by jplambert** [comment]
Hi and thanks for answering,

Indeed I forgot to replace "${jsonbody}" with "parseJSON(response.body)" -- I could not have "response.jsonbody" to work, but I'll try again, maybe I mistyped something. In any case it was just a variable defined in FitNesse so that writing the "let" part uses a similar syntax to the "GET" part.


I'll take a look at issue #52. I'll try to do something if it's not taking too much time.

As for extending RestFixture to calculate hash, I'd rather keep that outside of RestFixture as so many other processing could occur between two rest calls. That may be easier to implement than fixing issue #52, however, so that's still a fallback option.

Thanks again, I'll let you know about issue #52.


Oh and by the way, great job you did there with RestFixture! It works very well and brings a lot a value to test Rest API. The feature is quite mature, everything that needs to be configured can be. I also looked at the code, and it is very clean, it is very easy to navigate in it and understand the purpose and design.

Thanks again,

JP
